### ISSUE TITLE
Remove Password & Username in MOF

### ISSUE THREAD

**Issue by SahFari** [rubrikinc/rubrik-scripts-for-powershell#205]
Hi, 
I think it is best to not log Password & UserName in MOF. The user is sysadmin in SMSSQL & local admin on server

instance of MSFT_Credential as $MSFT_Credential1ref
{
Password = "";
 UserName = "";

};

Kind Regards,
/Fari

**Comment by clumnah** [comment]
I completely agree with you on this. I have reviewed this and after looking at 
https://devblogs.microsoft.com/powershell/want-to-secure-credentials-in-windows-powershell-desired-state-configuration/?msclkid=9a29771eaba911ec80af2096dbea18bf
and
https://docs.microsoft.com/en-us/powershell/dsc/pull-server/secureMOF?view=dsc-1.1&viewFallbackFrom=powershell-7.2

I cannot update the script to comply with the method that MSFT recommends. It would require the end user to have something in place prior to being able to run this script that is not going to be available in most environments. 

In the lab I am using for testing, I use Group Managed Service Accounts. While a password may be inputted, it is not the correct password as AD is setting it and using it for the service. The exposure is minimal in that case. 

If you have other ideas, I can try to incorporate them into the script.

**Comment by SahFari** [comment]
Hi,

Oh, I see.
Is it not possible to reset the password at the end of the script?

Regard,
Fari Sah

Fari Sah  |  IT-tekniker
Skogsstyrelsen  |  IT-enheten
55183 Jönköping
036-35 93 00 (växel)  |  036-35 94 31 (direkt)
***@***.******@***.***>  |  skogsstyrelsen.se<https://skogsstyrelsen.se/>

***@***.***

Från: Chris Lumnah ***@***.***>
Skickat: den 24 mars 2022 20:45
Till: rubrikinc/rubrik-scripts-for-powershell ***@***.***>
Kopia: Sah Fari IT-enheten ***@***.***>; Author ***@***.***>
Ämne: [EXTERNT] Re: [rubrikinc/rubrik-scripts-for-powershell] Remove Password & Username in MOF (Issue #205)


I completely agree with you on this. I have reviewed this and after looking at
https://devblogs.microsoft.com/powershell/want-to-secure-credentials-in-windows-powershell-desired-state-configuration/?msclkid=9a29771eaba911ec80af2096dbea18bf
and
https://docs.microsoft.com/en-us/powershell/dsc/pull-server/secureMOF?view=dsc-1.1&viewFallbackFrom=powershell-7.2

I cannot update the script to comply with the method that MSFT recommends. It would require the end user to have something in place prior to being able to run this script that is not going to be available in most environments.

In the lab I am using for testing, I use Group Managed Service Accounts. While a password may be inputted, it is not the correct password as AD is setting it and using it for the service. The exposure is minimal in that case.

If you have other ideas, I can try to incorporate them into the script.

**Comment by clumnah** [comment]
You could definitely do it a different way. For instance, you could do it via Set-Service see the link below.

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-service?msclkid=29123e70ac3411ec8cc661f4c8789b38&view=powershell-7.2https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-service?msclkid=29123e70ac3411ec8cc661f4c8789b38&view=powershell-7.2

I didn’t go this way because I wanted the declarative configuration of Desired State Configuration. It allowed for a cleaner installation, configuration and re-runnability. If I was to do this in my own environment where I had control over things, I would have taken the extra step of implementing the security certificates in the previous links. Doing that, would not allow me to publish on GitHub.

An alternative to my configuration is to Ansible. It is a similar declarative configuration, but you don’t need the extra certificate process to be implemented first. We use that in one of our labs and it is much cleaner than my approach as you can store credentials securely.

**Comment by SahFari** [comment]
Thanks Chris.

Regard,
Fari Sah

Fari Sah  |  IT-tekniker
Skogsstyrelsen  |  IT-enheten
55183 Jönköping
036-35 93 00 (växel)  |  036-35 94 31 (direkt)
***@***.******@***.***>  |  skogsstyrelsen.se<https://skogsstyrelsen.se/>

***@***.***

Från: Chris Lumnah ***@***.***>
Skickat: den 25 mars 2022 13:14
Till: rubrikinc/rubrik-scripts-for-powershell ***@***.***>
Kopia: Sah Fari IT-enheten ***@***.***>; Author ***@***.***>
Ämne: [EXTERNT] Re: [rubrikinc/rubrik-scripts-for-powershell] Remove Password & Username in MOF (Issue #205)

You could definitely do it a different way. For instance, you could do it via Set-Service see the link below.

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-service?msclkid=29123e70ac3411ec8cc661f4c8789b38&view=powershell-7.2https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-service?msclkid=29123e70ac3411ec8cc661f4c8789b38&view=powershell-7.2

I didn’t go this way because I wanted the declarative configuration of Desired State Configuration. It allowed for a cleaner installation, configuration and re-runnability. If I was to do this in my own environment where I had control over things, I would have taken the extra step of implementing the security certificates in the previous links. Doing that, would not allow me to publish on GitHub.

An alternative to my configuration is to Ansible. It is a similar declarative configuration, but you don’t need the extra certificate process to be implemented first. We use that in one of our labs and it is much cleaner than my approach as you can store credentials securely.
